### PR TITLE
Fix bug in retry code

### PR DESCRIPTION
**Pull request by davidfurey**
When the limiter code was added, a bug was introduced that causes the error `error:  ReferenceError: req is not defined` if a retry is attempted.

We can fix this by giving the limited function a name (`req`) and returning it.

Since fixing this bug would introduce unbounded retries, which might cause problems, I've also limited the number of retries to 3 attempts in total.

### CODE REVIEW

File: src/drive.js [hunk 1]
```diff
@@ -39,26 +39,28 @@ var request = (function() {
...
 
         try {
             return await rp({uri, 'headers': {'Authorization': `${token.token_type} ${token.access_token}`}});
         } catch (err) {
-            if (err.statusCode === 401) {
+            if (err.statusCode === 401 && maxRetries > 0) {
                 gu.log.info('Authorization token expired');
                 token = undefined;
-                return await req(uri);
+                await req(uri, maxRetries - 1);
```
Comment (jonathonherbert) [inline src/drive.js:52]: Do we want to return this result?

Comment (davidfurey) [reply src/drive.js:52]: We do!

**Review by jonathonherbert — APPROVED** [review]
Subtle function hoisting problem. `return function req` does not hoist `req` – declaring `function req` on another line and then returning it does. Looks good.

**Comment by davidfurey** [comment]
Testing in production now

**Pull request merged**